### PR TITLE
Redesign the network modifications in order to use powsybl modification class

### PR DESCRIPTION
**Pull request by SlimaneAmar**
Signed-off-by: Slimane AMAR <slimane.amar@rte-france.com>

### CODE REVIEW

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 1]
```diff
@@ -327,7 +327,7 @@ public ResponseEntity<Flux<ModificationInfos>> lineSplitWithVoltageLevel(@Parame
                                                                              @RequestParam(value = "group", required = false) UUID groupUuid,
                                                                              @RequestParam(value = "reportUuid") UUID reportUuid,
                                                                              @RequestBody LineSplitWithVoltageLevelInfos lineSplitWithVoltageLevelInfos) {
-        return ResponseEntity.ok().body(networkModificationService.splitLineWithVoltageLevel(networkUuid, variantId, groupUuid, reportUuid, lineSplitWithVoltageLevelInfos));
+        return ResponseEntity.ok().body(networkModificationService.splitLineWithVoltageLevel(networkUuid, variantId, new GroupAndReportInfos(groupUuid, reportUuid), lineSplitWithVoltageLevelInfos));
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:330]: yes

**Review by geofjamg — APPROVED** [review]
@SlimaneAmar much cleaner and extensible than main branch!

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/SplitLineWithVoltageLevel.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            modificationInfos.getPercent(),
+            voltageLeveId,
+            modificationInfos.getBbsOrBusId(),
+            modificationInfos.getNewLine1Id(),
+            modificationInfos.getNewLine1Name(),
+            modificationInfos.getNewLine2Id(),
+            modificationInfos.getNewLine2Name(),
+            line);
+
+        algo.apply(network, false, subReporter);
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/modifications/SplitLineWithVoltageLevel.java]: Done

File: src/main/resources/config/application.yaml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 spring:
   application:
     name: netork-modification-server
+  main:
+    allow-circular-references: true
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/resources/config/application.yaml]: Removed

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**